File: packages/amplify_api/ios/Classes/OperationsManager.swift
//
//  OperationsManager.swift
//  amplify_api
//
//  Created by Admin on 12/28/20.
//

import Foundation
import Amplify

public class OperationsManager{
    
    private static var operationsMap : [String:Operation] = [String:Operation]()
    
    public static func containsOperation(cancelToken : String) -> Bool{
        return operationsMap[cancelToken] != nil
    }
    
    public static func addOperation(cancelToken : String, operation : Operation){
        operationsMap[cancelToken] = operation
    }
    
    public static func removeOperation(cancelToken : String){
        if(containsOperation(cancelToken: cancelToken)){
            operationsMap.removeValue(forKey: cancelToken)
        }
    }
    
    public static func cancelOperation(cancelToken : String){
        operationsMap[cancelToken]?.cancel()
        removeOperation(cancelToken: cancelToken)
    }
}

File: packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/types/OperationsManager.kt
package com.amazonaws.amplify.amplify_api.types

import com.amplifyframework.core.async.Cancelable

class OperationsManager {

    companion object {

        private var operationsMap: HashMap<String, Cancelable> = HashMap()

        fun containsOperation(cancelToken: String): Boolean {
            return operationsMap.containsKey(cancelToken)
        }

        fun addOperation(cancelToken: String, operation: Cancelable) {
            operationsMap[cancelToken] = operation
        }

        fun removeOperation(cancelToken: String) {
            if(containsOperation(cancelToken)){
                operationsMap.remove(cancelToken)
            }
        }

        fun cancelOperation(cancelToken: String) {
            operationsMap[cancelToken]?.cancel()
            removeOperation(cancelToken)
        }

    }
}
